File: Projects/Mines_NetworkCall/Experiment/experiment.c
#include <stdio.h>
#include <RL_glue.h>

#define NUM_EPISODES 1000
int rl_num_steps[NUM_EPISODES];
double rl_return[NUM_EPISODES];

void run(int num_episodes)
{
/*run for num_episode number of episodes and store the number of steps and return from each episode*/        
  int x = 0;
  for(x = 0; x < num_episodes; ++x) {
    RL_episode(0);
    fprintf(stderr, ".");
    rl_num_steps[x] = RL_num_steps();
    rl_return[x] = RL_return();
  }
}

int main(int argc, char *argv[]) {
  unsigned int i = 0;
  double avg_steps = 0.0;
  double avg_return = 0.0;

/*basic main loop*/
  RL_init();
  run(NUM_EPISODES);
  RL_cleanup();
  
  /*add up all the steps and all the returns*/
  for (i = 0; i < NUM_EPISODES; i++) {
    avg_steps += rl_num_steps[i];
    avg_return += rl_return[i];
  }

/*average steps and returns*/
  avg_steps /= NUM_EPISODES;
  avg_return /= NUM_EPISODES;

/*print out results*/
  printf("\n-----------------------------------------------\n");
  printf("Number of episodes: %d\n",NUM_EPISODES);
  printf("Average number of steps per episode: %f\n", avg_steps);
  printf("Average return per episode: %f\n", avg_return);
  printf("-----------------------------------------------\n");
  
  return 0;
}

File: Tests/Network_Java/src/Experiment.java
import rlglue.RLGlue;
import rlglue.types.*;
import java.io.IOException;

public class Experiment
{
	protected static final int kNumEpisodes = 100;
	protected static int rlNumSteps[];
	protected static double rlReturn[];

	protected static void run(int numEpisodes) throws IOException
	{
		/*run for num_episode number of episodes and store the number of steps and return from each episode*/        	
		for(int x = 0; x < numEpisodes; ++x) {
			RLGlue.RL_episode(0);
			System.out.print(".");
			rlNumSteps[x] = RLGlue.RL_num_steps();
			rlReturn[x] = RLGlue.RL_return();
		}
	}

	public static void main(String [] args) throws IOException {
		double avgSteps = 0.0;
		double avgReturn = 0.0;

		rlNumSteps = new int[Experiment.kNumEpisodes];
		rlReturn = new double[Experiment.kNumEpisodes];

		System.err.println("+ RL_agent_message: query_agent");
		String agent_response = RLGlue.RL_agent_message("query_agent");
		System.err.println("- RL_agent_message: " + agent_response);
		
		System.err.println("+ RL_env_message: query_env");
		String env_response = RLGlue.RL_env_message("query_env");
		System.err.println("- RL_env_message: " + env_response);

		/*basic main loop*/

		for (int run = 0; run < 2; ++run)
		{
			RLGlue.RL_init();

			Random_seed_key key = new Random_seed_key(2, 2);
			for (int i = 0; i < 2; ++i)
			{
			    key.intArray[i] = i;
			    key.doubleArray[i] = (double) i;
			}
			RLGlue.RL_set_random_seed(key);

			String agentMessage = RLGlue.RL_agent_message("poke");
			System.out.println(agentMessage);

			String envMessage = RLGlue.RL_env_message("prod");
			System.out.println(envMessage);

			run(kNumEpisodes);
			RLGlue.RL_cleanup();

			/*add up all the steps and all the returns*/
			for (int i = 0; i < Experiment.kNumEpisodes; ++i) {
				avgSteps += rlNumSteps[i];
				avgReturn += rlReturn[i];
			}

			/*average steps and returns*/
			avgSteps /= (double)Experiment.kNumEpisodes;
			avgReturn /= (double)Experiment.kNumEpisodes;

			/*print out results*/
			System.out.println("\n-----------------------------------------------\n");
			System.out.println("Number of episodes: " + Experiment.kNumEpisodes);
			System.out.println("Average number of steps per episode: " +  avgSteps);
			System.out.println("Average return per episode: " + avgReturn);
			System.out.println("-----------------------------------------------\n");
		}   
	}
}
